### PR TITLE
fixes #371, fixes #236

### LINKED ISSUE #236: --mail options don't work
@kherner and @hschellman have both reported that the various --mail* options to `jobsub_submit` do not work.

### LINKED ISSUE #371: JOBSUBJOBSECTION is always 0 for DAG built out of maxconcurrent template
The env var `JOBSUBJOBSECTION` for worker jobs part of DAG built using the ` maxconcurrent` template is always set to `0`, instead of being a progressive index.
An example where this has been observed is the DAG submission `60725507@jobsub01.fnal.gov`.

For DAG built out of `dataset` template  `JOBSUBJOBSECTION` seems to be set properly.

### PR DESCRIPTION
**Pull request by marcmengel**
Just a few missing template lines....

### CODE REVIEW

**Review by shreyb — APPROVED** [review]
LGTM.

**Comment by shreyb** [comment]
Do we want to release this earlier than May 3?  Cut a 1.2.1 that has this, and then a 1.3 with token weakening?

File: templates/dag/dag.dag.condor.sub [hunk 1]
```diff
@@ -31,6 +31,10 @@ delegate_job_GSI_credentials_lifetime = 0
 {%endif%}
 {{lines|join("\n")}}
 
+{% if generate_email_summary %}
+notify= {{email_to}}
```
Comment (shreyb) [inline templates/dag/dag.dag.condor.sub:35]: Actually - hang on, I see other strings are quoted in the submission file.  Does this also need to be quoted?

**Review by shreyb — CHANGES_REQUESTED** [review]

File: templates/dag/dag.dag.condor.sub (hunk 1 above)
Comment (marcmengel) [reply templates/dag/dag.dag.condor.sub:35]: Yes, it probably should be.  Stuff sometimes kind of works when you forget, but...

File: templates/dag/dag.dag.condor.sub [hunk 2]
```diff
@@ -32,7 +32,8 @@ delegate_job_GSI_credentials_lifetime = 0
 {{lines|join("\n")}}
 
 {% if generate_email_summary %}
-notify= {{email_to}}
+notify = "{{email_to}}"
```
Comment (shreyb) [inline templates/dag/dag.dag.condor.sub:35]: Looking at the docs, shouldn't all of these `notify` entries be `notify_user`?

https://htcondor.readthedocs.io/en/latest/man-pages/condor_submit.html

File: templates/dag/dag.dag.condor.sub [hunk 3]
```diff
@@ -32,7 +32,8 @@ delegate_job_GSI_credentials_lifetime = 0
...
+notification = "always"
```
Comment (shreyb) [inline templates/dag/dag.dag.condor.sub:36]: Also, I don't know if condor submission file parsing is case-sensitive, but the same docs as above specify that the accepted value is "Always", not "always", so perhaps this should also be changed.

**Review by shreyb — CHANGES_REQUESTED** [review]
Couple of comments on "notify" vs "notify_user"

File: templates/dag/dag.dag.condor.sub (hunk 2 above)
Comment (marcmengel) [reply templates/dag/dag.dag.condor.sub]: Yes!  Thanks..

File: templates/dag/dag.dag.condor.sub (hunk 3 above)
Comment (marcmengel) [reply templates/dag/dag.dag.condor.sub]: Changed to upper-case throughout...

**Review by shreyb — COMMENTED** [review]
LGTM except for the config bit.

**Review by shreyb — APPROVED** [review]
Ran all unit tests on this PR, and they all pass. LGTM

**Pull request merged**